src/GraphQL/Conversion/ListConverterFactories/ArrayListConverterFactory.cs

```diff
@@ -0,0 +1,24 @@
+namespace GraphQL.Conversion;
+
+internal sealed class ArrayListConverterFactory : ListConverterFactoryBase
+{
+    private ArrayListConverterFactory()
+    {
+    }
+
+    public static ArrayListConverterFactory Instance { get; } = new ArrayListConverterFactory();
```

Comment:
Most of the internal factories do not have an `Instance` property, since they are only constructed once in the static constructor of `ValueConverter`.  Each of them should effectively be a singleton as they cache created list converters.

Another design would be to eliminate caching within the factories, and provide a global cache within `ValueConverter`.  So you call `ValueConverter.GetListConverter(Type listType)` and only incur a single dictionary lookup, rather than `ValueConverter.GetListConverterFactory(Type listType).Create(Type listType)` which incurs two dictionary lookups.